### ISSUE TITLE
pip3 installed software is not available in `PATH` in `macos-12`

### ISSUE THREAD

**Issue by general-kroll-4-life** [actions/runner-images#6496]
### Description

I am using `pip3` to install robot framework and other python packages.  In the `macos-11` image, they are all available to subsequent steps.  In the `macos-12` image I encountered, at least robot framework was not available.

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 10.15
- [ ] macOS 11
- [X] macOS 12
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

```
Current runner version: '2.298.2'
...
Image: macos-12
Version: 20221027.1
```
The below link is a private build to jog my memory, please see screen grabs also:

https://github.com/stackql/stackql-devel/actions/runs/3365385197/jobs/5580799140

![Screen Shot 2022-11-01 at 10 44 45 am](https://user-images.githubusercontent.com/82620104/199129969-b9c04e6d-de3d-41b5-aa77-e40303c99b0b.png)

![Screen Shot 2022-11-01 at 10 47 26 am](https://user-images.githubusercontent.com/82620104/199130073-76041e75-d41a-46ff-b07a-8199ff82b1ee.png)

### Is it regression?

No

### Expected behavior

Packages installed with `pip3` (eg robot framework) should be available to subsequent steps, as per `macos-11`.

### Actual behavior

Getting a `robot: command not found` error.

### Repro steps

Using the runner cited:

1. `pip3 install robotframework`
2. in a subsequent step `robot --version`

**Comment by igorboskovic3** [comment]
Hi @general-kroll-4-life, we will take a look.

**Comment by al-cheb** [comment]
Hey @general-kroll-4-life.

We have migrated python installation to use macOS universal package. Our recommendation is to use setup-python task or python3 -m syntax:

setup-python:
```
      - uses: actions/setup-python@v4
        with:
          python-version: '3.11' 
      - run: |
          pip3 install robotframework
      - run: |
          robot --version || true
      - run: |
          python3 -m robot --version || true
```
![image](https://user-images.githubusercontent.com/47745270/199199904-d05a709a-bb68-4b78-9676-545851f3a2aa.png)


https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#installing-python-on-macos
http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#return-codes
<html>
<body>


251 | Help or version information printed.
-- | --



</body>
</html>

python3 -m syntax:
```
      - run: |
          pip3 install robotframework
      - run: |
          python3 -m robot --version || true
```
![image](https://user-images.githubusercontent.com/47745270/199200062-ab323021-ed80-4d9b-b68c-a02ad221f5bf.png)

**Comment by general-kroll-4-life** [comment]
> Hey @general-kroll-4-life.
> 
> We have migrated python installation to use macOS universal package. Our recommendation is to use setup-python task or python3 -m syntax:
> 
> setup-python:
> 
> ```
>       - uses: actions/setup-python@v4
>         with:
>           python-version: '3.11' 
>       - run: |
>           pip3 install robotframework
>       - run: |
>           robot --version || true
>       - run: |
>           python3 -m robot --version || true
> ```
> 
> ![image](https://user-images.githubusercontent.com/47745270/199199904-d05a709a-bb68-4b78-9676-545851f3a2aa.png)
> 
> https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#installing-python-on-macos http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#return-codes
> 
> 251	Help or version information printed.
> python3 -m syntax:
> 
> ```
>       - run: |
>           pip3 install robotframework
>       - run: |
>           python3 -m robot --version || true
> ```
> 
> ![image](https://user-images.githubusercontent.com/47745270/199200062-ab323021-ed80-4d9b-b68c-a02ad221f5bf.png)

Cheers all

**Comment by al-cheb** [comment]
Close as by design.

**Comment by general-kroll-4-life** [comment]
@al-cheb this same thing is now happening in a `macos-11` runnner; is this intentional?

**Comment by al-cheb** [comment]
> @al-cheb this same thing is now happening in a `macos-11` runnner; is this intentional?

Yep, we use the same universal package.

**Comment by kleisauke** [comment]
Is this a duplicate of issue #6507, which was fixed with commit f6632ff53b24db9705c5b15bf90dbf05cc3fe295 (but not deployed at the time of writing)? If so, the workaround in comment https://github.com/actions/runner-images/issues/6507#issuecomment-1301636292 or waiting until the new macOS images are deployed would also fix this. :sweat_smile:
